torchci/lib/metricUtils.ts

```diff
@@ -0,0 +1,193 @@
+import { JobsPerCommitData, JobAnnotation } from "lib/types";
+
+// When N consecutive failures of the same type happen, the failures are counted as
+// broken trunk failures (approximately)
+export const BROKEN_TRUNK_THRESHOLD = 3;
+
+// When more than N failures happening in the same commit, the failures are counted
+// as part of an outage or broken infra (approximately)
+export const OUTAGE_THRESHOLD = 10;
+
+function getFailureByJobName(
+  jobName: string,
+  failures: { [jobName: string]: { [t: string]: number } }
+) {
+  if (!(jobName in failures)) {
+    failures[jobName] = {
+      [JobAnnotation.BROKEN_TRUNK]: 0,
+      [JobAnnotation.INFRA_BROKEN]: 0,
+      [JobAnnotation.TEST_FLAKE]: 0,
+    };
+  }
+
+  return failures[jobName];
+}
+
+function updateBrokenInfraCount(
```

Comment:
Naming nit, since you're incrementing the count instead of setting it

```suggestion
function incrementBrokenInfraCount(
```